src/lib/eina/eina_inline_lock_win32.x

```diff
@@ -524,90 +441,35 @@ _eina_tls_get(Eina_TLS key)
 static inline Eina_Bool
 _eina_tls_set(Eina_TLS key, const void *data)
 {
-   int ok = TlsSetValue(key, (void *) data);
-   DWORD err = GetLastError();
-   if (ok != 0 && err == ERROR_SUCCESS) return EINA_TRUE;
-   else return EINA_TRUE;
+   return TlsSetValue(key, (void *) data) ? EINA_TRUE : EINA_FALSE;
 }
 
-#ifdef EINA_HAVE_WIN32_BARRIER
 struct _Eina_Barrier
 {
-   LPSYNCHRONIZATION_BARRIER barrier;
+   SYNCHRONIZATION_BARRIER barrier;
 };
-#else
-struct _Eina_Barrier
-{
-   int needed;           /**< The number of waiting threads that will cause the barrier to signal and reset. */
-   int called;           /**< The number of threads that are waiting on this barrier. */
-   Eina_Lock cond_lock;  /**< The lock for the barrier */
-   Eina_Condition cond;  /**< The condition variable for the barrier */
-};
-#endif
 
 static inline Eina_Bool
 _eina_barrier_new(Eina_Barrier *barrier, int needed)
 {
-#ifdef EINA_HAVE_WIN32_BARRIER
-   InitializeSynchronizationBarrier(&(barrier->barrier), (LONG) needed, 0);
-   DWORD ok = GetLastError();
-   if (ok == ERROR_SUCCESS) return EINA_TRUE;
-   else if ((ok == ERROR_NOT_ENOUGH_MEMORY) || (ok == ERROR_ACCESS_DENIED)
-            || (ok == ERROR_OUTOFMEMORY))
-    {
-       return EINA_FALSE;
-    }
-   else EINA_LOCK_ABORT_DEBUG(ok, barrier_init, barrier);
-   return EINA_FALSE;
-#else
-   barrier->needed = needed;
-   barrier->called = 0;
-   if (eina_lock_new(&(barrier->cond_lock)))
-     {
-        if (eina_condition_new(&(barrier->cond), &(barrier->cond_lock)))
-          return EINA_TRUE;
-     }
-   return EINA_FALSE;
-#endif
+   return InitializeSynchronizationBarrier(&barrier->barrier, (LONG) needed, 10)
+       ? EINA_TRUE : EINA_FALSE;
 }
 
 static inline  void
 _eina_barrier_free(Eina_Barrier *barrier)
 {
-#ifdef EINA_HAVE_WIN32_BARRIER
-   DeleteSynchronizationBarrier(&(barrier->barrier));
-   DWORD ok = GetLastError();
-   if (ok != ERROR_SUCCESS) EINA_LOCK_ABORT_DEBUG(ok, barrier_destroy, barrier);
-#else
-   eina_condition_free(&(barrier->cond));
-   eina_lock_free(&(barrier->cond_lock));
-   barrier->needed = 0;
-   barrier->called = 0;
-#endif
+   if (!DeleteSynchronizationBarrier(&barrier->barrier))
+     {
+        EINA_LOCK_ABORT_DEBUG(GetLastError(), barrier_destroy, barrier);
```

Comment:
Why removing so many debugging info, but sticking with this one?

Reply:
The ones I removed are code paths no longer valid, but this is.